olive/model/__init__.py

```diff
@@ -454,9 +458,9 @@ def tensor_dtype_to_np_dtype(tensor_type):
                 io_config[f"{prefix}_shapes"].append(shape)
 
         # save io_config
-        self.io_config = io_config
+        self.io_config = IOConfig.parse_obj(io_config)
```

Comment:
> IOConfig.parse_obj(io_config) [](http://example.com/codeflow?start=24&length=30)

return dict instead of IOConfig will be much simpler. It is a standard behavior without needing know the definition.